### PR TITLE
Fixes crash for null ptr

### PR DESCRIPTION
**Pull request by NejcZdovc**
Resolves https://github.com/brave/brave-browser/issues/7363

It was introduced in here https://github.com/brave/brave-core/pull/4120

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- [ ] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protection-Mode
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Test Plan:
don't have STR, but this was found from the crash reports and what @emerick found as well

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: components/brave_rewards/browser/rewards_service_impl.cc [hunk 1]
```diff
@@ -1187,6 +1187,11 @@ void RewardsServiceImpl::LoadNicewareList(
 void RewardsServiceImpl::SavePublisherInfo(
     ledger::PublisherInfoPtr publisher_info,
     ledger::PublisherInfoCallback callback) {
+  if (!publisher_info) {
```
Comment (tmancey) [inline components/brave_rewards/browser/rewards_service_impl.cc]: Why are we getting `nullptr` in the first instance, at a minimum we should add a `DCHECK(publisher_info)` so we can track down the culprit

File: components/brave_rewards/browser/rewards_service_impl.cc [hunk 2]
```diff
@@ -1187,9 +1187,16 @@ void RewardsServiceImpl::LoadNicewareList(
...
+    callback(ledger::Result::LEDGER_ERROR, std::move(publisher_info));
+    return;
+  }
+
+  auto copy = publisher_info->Clone();
   base::PostTaskAndReplyWithResult(file_task_runner_.get(), FROM_HERE,
       base::BindOnce(&SavePublisherInfoOnFileTaskRunner,
-                    publisher_info->Clone(),
+                    std::move(copy),
```
Comment (tmancey) [inline components/brave_rewards/browser/rewards_service_impl.cc]: If a developer is copy/pasting there is potential for a bug, I would personnaly clone inline for each `std::bind`

Comment (NejcZdovc) [inline components/brave_rewards/browser/rewards_service_impl.cc]: this was the main problem, why we saw crashes. If checks are just additional safety

**Comment by NejcZdovc** [comment]
everything passed except Windows (build stopped), so restarting

**Pull request merged**